cfme/fixtures/vporizer.py

```diff
@@ -0,0 +1,101 @@
+from datetime import datetime
+import re
+from collections import namedtuple
+from random import random
+
+import pytest
+
+
+vpor_values_pattern = """---
+:avg:
+  :cpu_usagemhz_rate_average: {}
+  :derived_memory_used: {}
+  :max_cpu_usage_rate_average: {}
+  :max_mem_usage_absolute_average: {}
+:dev:
+  :cpu_usagemhz_rate_average: {}
+  :derived_memory_used: {}
+  :max_cpu_usage_rate_average: {}
+  :max_mem_usage_absolute_average: {}
+"""
+
+vpor_data_instance = namedtuple('vpor_data_instance',
+                                [
+                                    'resource_type',
+                                    'resource_id',
+                                    'resource_name',
+                                    'cpu_usagemhz_rate_average',
+                                    'derived_memory_used',
+                                    'max_cpu_usage_rate_average',
+                                    'max_mem_usage_absolute_average'
+                                ])
+
+cpu_usagemhz_rate_average__data_range = 8000
+derived_memory_used__data_range = 100
+max_cpu_usage_rate_average__data_range = 8000
+max_mem_usage_absolute_average__data_range = 100
+
+
+def gen_vpor_values():
+    return vpor_values_pattern.format(
+        random(), random() * cpu_usagemhz_rate_average__data_range,
+        random() * 100, random() * derived_memory_used__data_range,
+        random(), random() * max_cpu_usage_rate_average__data_range,
+        random() * 100, random() * max_mem_usage_absolute_average__data_range
+    )
+
+
+@pytest.yield_fixture(scope='module')
+def vporizer(appliance):
+
+    """Grabbing vim_performance_operating_ranges table data for nodes and projects.
+    In case that no such data exists, inserting fake rows"""
+
+    db = appliance.db
+    vpor = db['vim_performance_operating_ranges']
+
+    container_nodes = db['container_nodes']
+    container_pods = db['container_groups']
+    container_projects = db['container_projects']
+    created_at = datetime.now()
+    vpor_data_list = []
+
+    for table, resource_type in zip(
+        (container_nodes, container_projects, container_pods),
+        ('ContainerNode', 'ContainerProject', 'ContainerGroup')
+    ):
+
+        for resource in db.session.query(table).all():
+            def get_resource_vpor_data():
+                return vpor.__table__.select().where(
+                    (vpor.resource_id == resource.id) &
+                    (vpor.resource_type == resource_type)
+                ).execute().fetchone()
+            # Checking whether values has already inserted to the table for this resource.
+            # Should be True only in case that the appliance age is greater than 24h
+            # if there isn't such row, insert...
+            if not get_resource_vpor_data():
+                vpor.__table__.insert().values(
```

Comment:
Using "private" attributes raises questions - it cannot be done without using `__table__`?